### PR TITLE
Update requirements list

### PR DESCRIPTION
**Pull request by sharkykh**
**TL;DR** This doesn't affect any actual code. Safe to merge.

- Update the list to account for the changes made in #3877 
- Specify VCS sources for packages that are on a specific commit, and
- Enable VCS sources for some packages (mostly the unmodified ones)
- Pin `cfscrape` to <1.8.0 so it doesn't say "outdated" on requires.io
- Change `html5lib` version to 1.0b10 (new version style)
- Change `pgi` version to the actual commit hash

VCS sources won't work with setuptools (`setup.py`) in the current configuration,
but [there might be a possible solution](https://stackoverflow.com/a/18563923/7597273).

### CODE REVIEW

**Comment by sharkykh** [comment]
> VCS sources won't work with setuptools (setup.py) in the current configuration.

They actually do seem to work.. The SO answer I looked at was probably outdated.
